.github/workflows/build_workflow.yml

```diff
@@ -1,18 +1,44 @@
 name: Build workflow
 run-name: Build workflow
-on: [push]
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 jobs:
-  Explore-GitHub-Actions:
+  perl:
     runs-on: ubuntu-latest
     steps:
-      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
-      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by GitHub!"
-      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
-      - name: Check out repository code
-        uses: actions/checkout@v3
-      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
-      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
-      - name: List files in the repository
-        run: |
-          ls ${{ github.workspace }}
-      - run: echo "🍏 This job's status is ${{ job.status }}."
+      -
+        name: Checkout code
+        uses: actions/checkout@v4
+      -
+        name: Build Docker Image
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09
```

Comment:
May I know why you specifically choose this version of docker/build-push-action ? 
https://github.com/docker/build-push-action over here it shows the latest version is @v5 .

Reply:
security team suggested we use full length SHA commits when using third party actions. This SHA commit corresponds to the latest commit under the v5 tag.